app/services/recovery_code_generator.rb

```diff
@@ -8,15 +8,16 @@ def initialize(user, length: 16)
   end
 
   def create
-    @user_access_key = make_user_access_key(raw_recovery_code)
+    salt = Devise.friendly_token[0, 20]
+    @user_access_key = make_user_access_key(raw_recovery_code, salt)
 
-    user.update!(recovery_code: hashed_code)
+    user.update!(recovery_code: hashed_code, recovery_salt: salt)
```

Comment:
can we get some test coverage of this? specifically that generating a recovery code sets `user.recovery_salt` ?